source/documentation/tools/create-a-derived-table/style-guide.md

```diff
@@ -0,0 +1,415 @@
+    where code_set_name = 'REQUIREMENT TERMINATION REASON'
+),
+
+reference_4 as (
+    select * from {{ ref("derived_delius_stg__stg_standard_reference_4") }}
+    where code_set_name = 'UNITS'  
+),
+
+joined as (
+
+    select
+    requirements.rqmnt_id,
+    requirements.disposal_id,
+    requirements.start_date,
+    requirements.length,
+    requirements.rqmnt_notes,
+    requirements.commencement_date,
+    requirements.termination_date,
+    requirements.partition_area_id,
+    requirements.expected_start_date,
+    requirements.soft_deleted,
+    requirements.expected_end_date,
+
+    -- For rqmnt main type sub-cat
+    reference_1.code_value as rqmnt_type_sub_category_code,
+    reference_1.code_description as rqmnt_type_sub_category_desc,
+
+    -- For rqmnt main type sub-cat
+    reference_2.code_value as ad_rqmnt_type_sub_category_code,
+    reference_2.code_description as ad_rqmnt_type_sub_category_desc,
+
+    -- For terminations
+    reference_3.code_value as rqmnt_termination_reason_code,
+    reference_3.code_description as rqmnt_termination_reason_desc,
+
+    -- For units, bring code_value and code_description from standard_reference_list_id
+    reference_4.code_value as rqmnt_type_main_units_code,
+    reference_4.code_description as rqmnt_type_main_units_desc
+
+    from requirements
+
+    left join reference_1
+        on requirements.rqmnt_type_sub_category_id = reference_1.standard_reference_list_id
+
+    left join reference_2
+        on requirements.ad_rqmnt_type_sub_category_id = reference_2.standard_reference_list_id
+
+    left join reference_3
+        on requirements.rqmnt_termination_reason_id = reference_3.standard_reference_list_id
+
+    left join reference_4
+        on requirements.units_id = reference_4.standard_reference_list_id
+)
+
+select * from joined
+```
+
+## 'Functional' CTEs
+
+- ☝🏻 Where performance permits, CTEs should perform a single, logical unit of work.
+- 👭🏻 Where possible joins should be done separately to calculations. This is to make the code neater, we expect the table each columns comes from to be referenced and therefore can make calculations overly complicated if a join is also being performed. (see example at the end of the section)
+- 📖 CTE names should be as verbose as needed to convey what they do e.g. `events_joined_to_users` instead of `user_events` (this could be a good model name, but does not describe a specific function or transformation).
+- 🌉 CTEs that are duplicated across models should be pulled out into their own intermediate models. Look out for chunks of repeated logic that should be refactored into their own model.
+- 🔚 The last line of a model should be a `select *` from your final output CTE. This makes it easy to materialise and audit the output from different steps in the model as you're developing it. You just change the CTE referenced in the `select` statement to see the output from that step.
```

Comment:
when using CTEs

Reply:
Do you think there are examples not using CTEs?

Reply:
i think you can ignore my comment as it's actually clear you're talking about using CTEs